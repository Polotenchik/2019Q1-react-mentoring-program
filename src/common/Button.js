import React, {Component} from 'react';

export const Button = (props) => {
	return (
		<div className={
			(props.type === 'header' && 'btn btn-white')
			|| ( props.type === 'search' && 'btn btn-red') 
			|| ( props.type === 'sort' && 'btn btn-grey') 
			|| null
		}>
			{ props.content }
		</div>
	);
};
