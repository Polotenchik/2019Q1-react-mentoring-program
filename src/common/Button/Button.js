import React, {Component} from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

export const Button = (props) => {

	let itemClass = classNames({
		'btn': true,
		'btn-white': props.btnWhite,
		'btn-red': props.btnRed,
		'btn-grey': props.btnGrey,
	  });

	return (
		<div className={
			(props.type === 'header' && itemClass)
			|| ( props.type === 'search' && itemClass) 
			|| ( props.type === 'sort' && itemClass) 
			|| null
		}>
			{ props.content }
		</div>
	);
};

Button.propTypes = {
	btnWhite: PropTypes.bool,
	btnRed: PropTypes.bool,
	btnGrey: PropTypes.bool,
	content: PropTypes.string,
	id: PropTypes.string
};
  
Button.defaultProps = {
	btnWhite: false,
	btnRed: false,
	btnGrey: false,
	content: 'Button',
	id: ''
};