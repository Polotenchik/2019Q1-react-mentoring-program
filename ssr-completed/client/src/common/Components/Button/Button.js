import styled, { css } from 'styled-components';
import Theme from '../../Theme/Theme';

const Button = styled.button`
    font-size: 16px;
    margin-left: auto;    
    border: 0;
    padding: 6px 12px;    
    text-transform: uppercase;
    border-radius: 4px;
    outline: none;
    cursor: pointer;
    
    ${props => props.red && css`
        background-color: ${Theme.colors.red};
        color: ${Theme.colors.white};
    `}
    
    ${props => props.white && css`
        background-color: ${Theme.colors.white};
        color: ${Theme.colors.red};        
    `}
`;

export default Button;
