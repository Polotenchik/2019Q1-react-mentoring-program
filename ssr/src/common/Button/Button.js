import React, {Component} from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

export class Button extends Component {
    constructor(props) {
      super(props);

	}
	
    onClickFunc = () => {
      this.props.btnClick(this.props.content);
	};
	
    render() {
    const {btnSm, btnBig, btnWhite, btnRed, btnGrey, btnBold, btnTransp, btnTranspRed, btnUC} = this.props;
    const {id, content} = this.props;
    let itemClass = classNames({
      'btn': true,
      'btn-sm': btnSm,
      'btn-big': btnBig,
      'btn-white': btnWhite,
      'btn-red': btnRed,
      'btn-grey': btnGrey,
      'btn-bold':  btnBold,
      'btn-transp': btnTransp,
      'btn-transp-red': btnTranspRed,
      'btn-UC': btnUC
	});
	
    return ( 
		<div className={ itemClass } onClick={ this.onClickFunc } id={ id }>
      		{ content }
		</div>
	);
  }
}

Button.defaultProps = {
  btnSm: false,
  btnBig: false,
  btnWhite: false,
  btnRed: false,
  btnGrey: false,
  btnBold: false,
  btnTransp: false,
  btnTranspRed: false,
  btnUC: false,
  content: 'Button',
  id: ''
};