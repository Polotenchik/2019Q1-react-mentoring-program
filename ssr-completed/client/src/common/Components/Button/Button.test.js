import React from 'react';
import { mount} from 'enzyme';
import Button from './Button';

describe('Button Snapshot', () => {
    it('Button Common Component should render correctly with no props', () => {
        const component = mount(
            <Button modifiers={['red']}>Push</Button>
        );
        expect(component).toMatchSnapshot();
    });
});