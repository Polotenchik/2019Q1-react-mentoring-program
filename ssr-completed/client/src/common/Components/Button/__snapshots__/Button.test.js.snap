// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot Button Common Component should render correctly with no props 1`] = `
<styled.button
  modifiers={
    Array [
      "red",
    ]
  }
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "iYlrHO",
          "rules": Array [
            "
    font-size: 16px;
    margin-left: auto;    
    border: 0;
    padding: 6px 12px;    
    text-transform: uppercase;
    border-radius: 4px;
    outline: none;
    cursor: pointer;
    
    ",
            [Function],
            "
    
    ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    modifiers={
      Array [
        "red",
      ]
    }
  >
    <button
      className="sc-bdVaJa iYlrHO"
    >
      Push
    </button>
  </StyledComponent>
</styled.button>
`;
